datafusion/core/src/datasource/physical_plan/parquet/row_groups.rs

```diff
@@ -36,58 +36,34 @@ use crate::datasource::physical_plan::parquet::statistics::{
 };
 use crate::physical_optimizer::pruning::{PruningPredicate, PruningStatistics};
 
-use super::ParquetFileMetrics;
+use super::{ParquetAccessPlan, ParquetFileMetrics};
 
-/// Tracks which RowGroups within a parquet file should be scanned.
+/// Calculates which RowGroups within a parquet file should be scanned.
 ///
-/// This struct encapsulates the various types of pruning that can be applied to
+/// This struct implements the various types of pruning that are applied to
 /// a set of row groups within a parquet file, progressively narrowing down the
 /// set of row groups that should be scanned.
-#[derive(Debug, PartialEq)]
-pub struct RowGroupSet {
-    /// `row_groups[i]` is true if the i-th row group should be scanned
-    row_groups: Vec<bool>,
+#[derive(Debug, Clone, PartialEq)]
```

Comment:
this extracts the plan of what to scan from the logic that prunes it based on row group statistic metadat